### DISCUSSION TITLE
Can we have MIT license code from other developers in our own repositories

### DISCUSSION THREAD

**Discussion by dcorralf** [community/community#23453] [New to GitHub]
<p>Hi guys,</p>
<p>I’m wondering if in our own repositories, can we have other developers code (with MIT license for example or Apache) ?. Of course keeping the same original license. Thanks in advance !</p>
<p>Greetings.</p>

**Comment by jfullstackdev** [comment]
<p>as always, disclaimer first, even GitHub has the disclaimer, this is not a legal advice just my own opinion</p>
<aside class="quote no-group" data-username="dcorralf" data-post="1" data-topic="246972">
<div class="title">
<div class="quote-controls"></div>
 dcorralf:</div>
<blockquote>
<p>I’m wondering if in our own repositories, can we have other developers code (with MIT license for example or Apache) ?</p>
</blockquote>
</aside>
<p>if say the project is based from another project, and you and other developers have the modification, as long as you satisfied the conditions of the first license, I think, it’s your turn to decide what will be the next license for each modification you selected,</p>
<p>unless the original owners of the project prohibits you to have any other license than the original,</p>

**Comment by dcorralf** [comment]
<aside class="quote no-group" data-username="jdevstatic" data-post="2" data-topic="246972">
<div class="title">
<div class="quote-controls"></div>
 jdevstatic:</div>
<blockquote>
<p>if say the project is based from another project, and you and other developers have the modification, as long as you satisfied the conditions of the first license, I think, it’s your turn to decide what will be the next license for each modification you selected,</p>
</blockquote>
</aside>
<p>Hi, thanks for the reply.</p>
<p>My intention is to only use that code, maybe make a small modification in some cases (not in all), but my idea is to keep the original license, for example MIT. But I would like to copy the code of other developers in my personal Github repository, but I repeat, always keeping the original license (most of the code that I am going to copy is under the MIT license). I can do this, right?</p>
<p>Greetings.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The <a href="https://choosealicense.com/licenses/mit/">MIT license</a> is pretty direct about the permissions granted (emphasis added), and the second paragraph lists the conditions under which you receive those permissions:</p>
<blockquote>
<p><strong>Permission is hereby granted,</strong> free of charge, to any person obtaining a copy of this software and associated documentation files (the “Software”), <strong>to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software</strong>, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:</p>
<p>The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.</p>
<p>[…]</p>
</blockquote>
<p>I am also not a lawyer, but my opinion is that what you want do falls pretty clearly under “use, copy, modify, merge, publish, distribute”. :smiley_cat:</p>

**Reply by JellyLion** [reply]
But, for example, if I'm using MIT licensed code in my app, there exactly do i include this copyright notice and permission? Do i include it directly in the code as a comment or do I add the MIT license and specify what code was gotten from the original Software? This I ask because I would like to use another license instead of MIT.

**Reply by airtower-luna** [reply]
> But, for example, if I'm using MIT licensed code in my app, there exactly do i include this copyright notice and permission?

If you mean in the UI at runtime, a common approach is to have a list of used components and licenses available somewhere, often via or next to an "About" thing (e.g. Firefox has `about:license`, there you also have multiple licenses for different components).

> Do i include it directly in the code as a comment or do I add the MIT license and specify what code was gotten from the original Software?

In the source code I'd make sure to keep the notice as is, as the license text says: „shall be included in all copies“. :wink: There may or may not be some wiggle room with things like SPDX-identifiers or provenance descriptions, but I'd be very careful about *replacing* the original notice, and why make things more complicated than necessary?

**Reply by JellyLion** [reply]
Right, I see, is there any actual way to know whether you are doing it correctly? Thanks a lot!

**Reply by airtower-luna** [reply]
The proper answer is: Ask a copyright lawyer who's familiar with Open Source licenses.

The next best thing is to follow common practice in the community, looking at well established projects (e.g. Linux kernel, distros are a good example too as they package other people's source all the time) and how they do it. As long as you're acting in good faith (this includes definitely not removing any copyright notices!) I'd say it's unlikely someone will get angry enough to do more than send you a mail asking to fix some notice or similar (assuming you then do that).

**Comment by dcorralf** [comment]
<p>Great, I hadn’t read the license conditions and I see that they are very clear about what I want to do. You are absolutely right, according to the terms I think I can obtain copies without problems and even have it in my Github repository, always keeping the same license and copyright notice.<br>
Thank you very much, I think you have given me the answer to my questions.</p>
<p>Greetings.</p>

**Reply by userdrizzle** [reply]
what code have you found? what are you trying to do?

**Comment by bluesky0341** [comment]
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/zblogcn/zblogphp">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/zblogcn/zblogphp" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/105836139/181077187-9feb15a6-474f-4af8-9270-42fa648a19c8.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/9/497f5b3edd4212d9c91809e689cc0c02cb778290_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/4/9/497f5b3edd4212d9c91809e689cc0c02cb778290_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/4/9/497f5b3edd4212d9c91809e689cc0c02cb778290.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/9/497f5b3edd4212d9c91809e689cc0c02cb778290_2_10x10.png"></div>

<h3><a href="https://github.com/zblogcn/zblogphp" target="_blank" rel="noopener nofollow ugc">GitHub - zblogcn/zblogphp: Z-BlogPHP博客程序</a></h3>

  <p>Z-BlogPHP博客程序. Contribute to zblogcn/zblogphp development by creating an account on GitHub.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>above is MIT</p>
<p>but i have found that this company charge money to user for commercial use.one website is USD190<br>
<a href="https://www.zblogcn.com/service/" class="onebox" target="_blank" rel="noopener nofollow ugc">https://www.zblogcn.com/service/</a></p>

**Reply by Xcopz1** [reply]
33BRJWEKA5J553C7TQRIZ7AMVCPTYI43
